pkg/cmd/pulumi/destroy.go

```diff
@@ -149,6 +150,34 @@ func newDestroyCmd() *cobra.Command {
 			if err != nil {
 				return result.FromError(err)
 			}
+
+			var protectedCount int
+			if excludeProtected {
+				snapshot, err := s.Snapshot(commandContext())
+				if err != nil {
+					return result.FromError(err)
+				} else if snapshot == nil {
+					return result.FromError(errors.New("Failed to find the stack snapshot. Are you in a stack?"))
+				}
+				var unprotected []*resource.State
+				unprotected, protected := seperateProtected(snapshot.Resources)
+				protectedCount = len(protected)
+				if len(unprotected) == 0 && protectedCount > 0 {
+					fmt.Printf("There were no unprotected resources to destroy. There are still %d"+
+						" protected resources associated with this stack.\n", protectedCount)
+					// We need to return now. Otherwise the update will conclude
+					// we tried to destroy everything and error for trying to
+					// destroy a protected resource.
+					return nil
+				}
+				for _, r := range unprotected {
+					targetUrns = append(targetUrns, r.URN)
+				}
+			}
+
+			if targets != nil && len(*targets) > 0 && excludeProtected {
+				return result.FromError(errors.New("You cannot specify --target and --exclude-protected"))
```

Comment:
Cannot specify it "yet"? Is there a fundamental problem here?

Reply:
`--exclude-protected` means destroy everything that is not protected. `--target` means destroy this specific thing. Any flag specified by `--target` would either be invalid (because it is protected) or would already be targeted by the destroy. 

Because these flag behaviors don't make sense together, I forbid specifying both of them.